ipynb/examples/utils/gem5_integration.ipynb

```diff
@@ -0,0 +1,342 @@
+    "            # Disk of the distrib to run\n",
+    "            \"disk\"     : \"/home/user/Work/gem5/platform_juno/aarch64-ubuntu-trusty-headless.img\",\n",
+    "        },\n",
+    "        # gem5 settings\n",
+    "        \"simulator\" : {\n",
+    "            # Path to gem5 binary\n",
+    "            \"gem5_bin\"     : \"/home/user/Work/gem5/gem5.fast\",\n",
+    "            # Args to be given to the binary\n",
+    "            \"args\" : [\n",
+    "                # Zilch\n",
+    "            ],\n",
+    "        }\n",
+    "    },\n",
+    "\n",
+    "    # FTrace events to collect for all the tests configuration which have\n",
+    "    # the \"ftrace\" flag enabled\n",
+    "    \"ftrace\"  : {\n",
+    "         \"events\" : [\n",
+    "            \"sched_switch\",\n",
+    "            \"sched_wakeup\",\n",
+    "            \"sched_overutilized\",\n",
+    "            \"sched_load_avg_cpu\",\n",
+    "            \"sched_load_avg_task\",\n",
+    "            \"sched_load_waking_task\",\n",
+    "            \"cpu_capacity\",\n",
+    "            \"cpu_frequency\",\n",
+    "            \"cpu_idle\",\n",
+    "            \"sched_energy_diff\"\n",
+    "        ],\n",
+    "        \"buffsize\" : 100 * 1024,\n",
+    "    },\n",
+    "\n",
+    "    # Tools required by the experiments\n",
+    "    \"tools\"   : ['trace-cmd'],\n",
+    "    \n",
+    "    # Output directory on host\n",
+    "    \"results_dir\" : \"gem5_res\"\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true,
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# Create the hardware target. Patience is required :\n",
+    "# ~40 minutes to resume from a checkpoint\n",
+    "# ~3 hours to start from scratch\n",
+    "te = TestEnv(conf)\n",
+    "target = te.target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# Running some binaries on gem5"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "# [binary/script, arguments]\n",
+    "commands = [\n",
+    "    ['sysbench', '--test=cpu --max-time=1 run']\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "for command, args in commands:\n",
+    "    print \"-----[\" + command + \"]-----\"\n",
+    "    \n",
+    "    path = target.install_if_needed(\"/home/vagrant/lisa/tools/arm64/\" + command)\n",
+    "    print path\n",
+    "    \n",
+    "    target.execute('m5 resetstats')\n",
```

Comment:
It might be worth adding a link here to http://gem5.org/M5ops to show different arguments of m5.